pcdscalc/be_lens_calcs.py

```diff
@@ -0,0 +1,732 @@
+
+    transmission_total = calc_trans_for_single_lens(energy, radius_total,
+                                                    material, density,
+                                                    fwhm_unfocused,
+                                                    equivalent_disk_thickness,
+                                                    apex_distance_tot)
+    return transmission_total
+
+
+def calc_lens_set(energy, size_fwhm, distance, n_max=12, max_each=5,
+                  lens_radii=[100e-6, 200e-6, 300e-6, 500e-6, 1000e-6],
+                  fwhm_unfocused=0.0005, eff_rad0=None):
+    '''
+    Calculate lens set
+
+    Parameters
+    ----------
+    energy : number
+        Beam Energy
+    size_fwhm : `float`
+    distance : `float`
+    n_max : `int`
+    max_each : `int`
+    lens_radii : `list`
+    fwhm_unfocused : `float`
+        This is about 400 microns at XPP. Default = 0.0005
+    eff_rad0 : TODO: what is it?
+
+    Returns
+    -------
+    TODO: what returns
+
+    '''
+    nums = product(*([list(range(max_each + 1))] * len(lens_radii)))
+    sets = []
+    sizes = []
+    effrads = []
+    foclens = []
+    for num in nums:
+        lens_set = []
+        if sum(num) <= n_max and sum(num) > 0:
+            if eff_rad0 is None:
+                teffradinv = 0
+            else:
+                teffradinv = 1 / eff_rad0
+            for tn, tl in zip(num, lens_radii):
+                lens_set += [tn, tl]
+                teffradinv += tn / tl
+            teffrad = np.round(1 / teffradinv, 6)
+            if teffrad in effrads:
+                ind = effrads.index(teffrad)
+                if sum(sets[ind]) > sum(num):
+                    sets[ind] = num
+                else:
+                    continue
+            elif teffrad is not None:
+                effrads.append(teffrad)
+                sets.append(num)
+                size_fwhm = calc_beam_fwhm(energy, lens_set + [1, eff_rad0],
+                                           distance=distance,
+                                           source_distance=None,
+                                           fwhm_unfocused=fwhm_unfocused,
+                                           printsummary=False)
+                sizes.append(size_fwhm)
+                focal_length = calc_focal_length(energy,
+                                                 lens_set + [1, eff_rad0])
+                foclens.append(focal_length)
+
+    sizes = np.asarray(sizes)
+    sets = np.asarray(sets)
+    foclens = np.asarray(foclens)
+    indsort = (np.abs(sizes - size_fwhm)).argsort()
+
+    lens_sets = (sets[indsort, :],
+                 np.asarray(effrads)[indsort],
+                 sizes[indsort],
+                 foclens[indsort])
+
+    return lens_sets
+
+
+# TODO: ========== WE MIGHT NOT NEED THESE FUNCTIONS BELOW =================
+# TODO: distance default might need to be changed
+def find_radius(energy, distance=4.0, material="Be", density=None):
```

Comment:
might not need `find_radius` here?  - we had it in the old code

Reply:
Can leave it in for now, I suppose, since you've already done the work here.